packages/api/app/distribution-routes.js

```diff
@@ -44,6 +45,34 @@ function isPublicRequest(path) {
   }
 }
 
+const isTokenAuthRequest = (req) =>
+  req.get('EDL-Client-Id') && req.get('EDL-Token');
+
+const handleTokenAuthRequest = async (req, res, next) => {
+  try {
+    const userName = await req.earthdataLoginClient.getTokenUsername({
```

Comment:
It's confusing how this `req.earthdataLoginClient`  works here

Reply:
For Cognito, we use getUserInfo to get username and other information

Reply:
I think we should try to make our distribution api works for `OAuth-Client-Id` and `OAuth-Token` for both EDL and Cognito

Reply:
I agree, Don't know if this is necessary